monix-execution/shared/src/main/scala/monix/execution/FutureUtils.scala

```diff
@@ -161,6 +172,10 @@ object FutureUtils {
     p.future
   }
 
+  /** Creates a future that never completes. */
+  def never[A]: Future[A] =
+    Promise[A].future
```

Comment:
But then `never` from Scala 2.12 overrides everything else, e.g. `flatMap`.

@ktoso If we aren't overriding the other methods like`flatMap`, I think that represents a potential memory leak as well.

@larsrh we might need to do a Scala 2.11 specific object (in a `scala_2.11` directory) and just rely on Scala's `never` for 2.12.